### PR TITLE
utils: add ip address related functions

### PR DESCRIPTION
**Pull request by derekbit**
Longhorn/longhorn#6138

### CODE REVIEW

File: utils/ip.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		}
+	}
+	if ipv4 == nil {
+		return "", errors.Errorf("interface %s don't have an IPv4 address\n", name)
+	}
+
+	return ipv4.String(), nil
+}
+
+// GetIPForPod returns the IP address for the pod.
```
Comment (innobead) [inline utils/ip.go:43]: Suggest `// GetIPForPod returns the IP address for the pod from the storage network first or the cluster network.`

File: utils/ip.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	for _, ip := range ips {
+		if !ip.IsLoopback() {
+			return false
+		}
+	}
+	return true
+}
+
+// GetExternalIP returns the external IP address.
+func GetExternalIP() (string, error) {
```
Comment (innobead) [inline utils/ip.go:78]: It seems getting any external IP. Suggest `GetAnyExternalIP`.

File: utils/ip_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import (
+	"net"
+	"strings"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/longhorn/go-common-libs/test"
+)
+
+func (s *TestSuite) GetLocalIPv4fromInterface(c *C) {
```
Comment (innobead) [inline utils/ip_test.go:12]: The test function name is incorrect. Should be `TestIsLoopbackHost`

**Review by innobead — CHANGES_REQUESTED** [review]

**Comment by derekbit** [comment]
All are addressed. Thank you.

File: utils/ip_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+import (
+	"net"
+	"strings"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/longhorn/go-common-libs/test"
+)
+
+func (s *TestSuite) GetIsLoopbackHost(c *C) {
```
Comment (innobead) [inline utils/ip_test.go:12]: This should be "TestIsLoopbackHost"?

**Review by innobead — CHANGES_REQUESTED** [review]

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@derekbit moved ip related code to the `net` package.

File: utils/ip_test.go (hunk 4 above)
Comment (derekbit) [reply utils/ip_test.go]: Ah, forgot prefix..

**Pull request merged**